plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -50,13 +52,171 @@ type CiscoTelemetryMDT struct {
 
 	// Internal state
 	aliases   map[string]string
+        dmes      map[string]string
 	warned    map[string]struct{}
 	extraTags map[string]map[string]struct{}
+	nxpathMap map[string]map[string]string //per path map
+	nxpropMap map[string]int //Global property map
 	mutex     sync.Mutex
 	acc       telegraf.Accumulator
 	wg        sync.WaitGroup
 }
 
+type JsonStructure struct {
+	Name string `json:"Name"`
+	Prop []struct {
+		Key   string `json:"Key"`
+		Value string `json:"Value"`
+	} `json:"prop"`
+}
+
+//xform Field to string
+func xformValueString(field *telemetry.TelemetryField) interface{} {
+	var str string
+        switch val := field.ValueByType.(type) {
+        case *telemetry.TelemetryField_StringValue:
+                if len(val.StringValue) > 0 {
+                        return val.StringValue
+                }
+        case *telemetry.TelemetryField_Uint32Value:
+		str = strconv.FormatUint(uint64(val.Uint32Value), 10)
+		return str
+        case *telemetry.TelemetryField_Uint64Value:
+		str = strconv.FormatUint(val.Uint64Value, 10)
+                return str
+        case *telemetry.TelemetryField_Sint32Value:
+		str = strconv.FormatInt(int64(val.Sint32Value), 10)
+		return str
+        case *telemetry.TelemetryField_Sint64Value:
+		str = strconv.FormatInt(val.Sint64Value, 10)
+		return str
+        }
+        return nil
+}
+
+//Xform value field.
+func (c *CiscoTelemetryMDT) nxosValueXform(field *telemetry.TelemetryField, value interface{}, path string) interface{} {
+     isNXOS := !strings.ContainsRune(path, ':')
+     if (isNXOS) {
+        if xfrom, ok := c.nxpropMap[field.Name]; ok {
+           if xfrom == 1 {
+               switch field.ValueByType.(type) {
+               case *telemetry.TelemetryField_Uint64Value:
+                    return int64(value.(uint64))
+               default:
+                    return nil
+               }
```

Comment:
No need for using a switch, a simple if should suffice. Furthermore, why don't you directly type-assert on value? It seems a bit fragile to blindly assume that `value` always follows the type specified in `field`...
Furthermore, you risk wrapping of values here by casting your `uin64`  to `int64`.